src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1014,7 +1014,16 @@ public new CSharpCompilation ReplaceSyntaxTree(SyntaxTree oldTree, SyntaxTree? n
         internal override int GetSyntaxTreeOrdinal(SyntaxTree tree)
         {
             Debug.Assert(this.ContainsSyntaxTree(tree));
-            return _syntaxAndDeclarations.GetLazyState().OrdinalMap[tree];
+            try
+            {
+                return _syntaxAndDeclarations.GetLazyState().OrdinalMap[tree];
+            }
+            catch (KeyNotFoundException)
+            {
+                // Explicitly catching and re-throwing exception so we don't send the syntax
+                // tree (potentially containing private user information) to telemetry.
+                throw new KeyNotFoundException("Syntax tree not found.");
```

Comment:
this won't change anything.  If a dump is collected, then the data about everything in the user's memory space will be included.

Reply:
This seems overly aggressive anyways: I can imagine not wanting to include tree contents, but can we still include the path?